src/sentry/integrations/utils/sync.py

```diff
@@ -77,48 +81,52 @@ def sync_group_assignee_inbound(
-        "email": email,
-        "issue_key": external_issue_key,
-    }
-    if not affected_groups:
-        logger.info("no-affected-groups", extra=log_context)
-        return []
-
-    if not assign:
+    with ProjectManagementEvent(
+        action_type=ProjectManagementActionType.INBOUND_ASSIGNMENT_SYNC, integration=integration
+    ).capture() as lifecycle:
+        orgs_with_sync_enabled = where_should_sync(integration, "inbound_assignee")
+        affected_groups = Group.objects.get_groups_by_external_issue(
+            integration,
+            orgs_with_sync_enabled,
+            external_issue_key,
+        )
+        log_context = {
+            "integration_id": integration.id,
+            "email": email,
+            "issue_key": external_issue_key,
+        }
+        if not affected_groups:
+            logger.info("no-affected-groups", extra=log_context)
+            return []
+
+        if not assign:
+            for group in affected_groups:
+                GroupAssignee.objects.deassign(
+                    group,
+                    assignment_source=AssignmentSource.from_integration(integration),
+                )
+
+            return affected_groups
+
+        users = user_service.get_many_by_email(emails=[email], is_verified=True)
+        users_by_id = {user.id: user for user in users}
+        projects_by_user = Project.objects.get_by_users(users)
+
+        groups_assigned = []
         for group in affected_groups:
-            GroupAssignee.objects.deassign(
-                group,
-                assignment_source=AssignmentSource.from_integration(integration),
-            )
-
-        return affected_groups
-
-    users = user_service.get_many_by_email(emails=[email], is_verified=True)
-    users_by_id = {user.id: user for user in users}
-    projects_by_user = Project.objects.get_by_users(users)
-
-    groups_assigned = []
-    for group in affected_groups:
-        user_id = get_user_id(projects_by_user, group)
-        user = users_by_id.get(user_id)
-        if user:
-            GroupAssignee.objects.assign(
-                group,
-                user,
-                assignment_source=AssignmentSource.from_integration(integration),
-            )
-            groups_assigned.append(group)
-        else:
-            logger.info("assignee-not-found-inbound", extra=log_context)
-    return groups_assigned
+            user_id = get_user_id(projects_by_user, group)
+            user = users_by_id.get(user_id)
+            if user:
+                GroupAssignee.objects.assign(
+                    group,
+                    user,
+                    assignment_source=AssignmentSource.from_integration(integration),
+                )
+                groups_assigned.append(group)
+            else:
+                lifecycle.record_halt("inbound-assignee-not-found", extra=log_context)
+                logger.info("inbound-assignee-not-found", extra=log_context)
```

Comment:
we should save a cleanup task for ourselves to remove these redundant loggers

Reply:
Good call. Adding it to our SLOs tasks